xdsl/transforms/loop_invariant_code_motion.py

```diff
@@ -32,9 +41,82 @@ def can_be_hoisted(op: Operation, target_region: Region) -> bool | None:
     for child in op.walk():
         for operand in child.operands:
             operand_owner = operand.owner
-            assert operand_owner is not None
             if op.is_ancestor(operand_owner):
                 continue
             if target_region.is_ancestor(operand_owner):
                 return False
     return True
+
+
+def _move_loop_invariant_code(
+    regions: Sequence[Region],
+    should_move_out_of_region: Callable[[Operation, Region], bool],
+    move_out_of_region: Callable[[Operation, Region], None],
+):
```

Comment:
I would also return `None` (I'm not sure why we return the number of operation moved), or mark explicitely that the return value is `int` (and document that)

Reply:
Yeah I'll change to None

Reply:
Done